src/symbols.js

```diff
@@ -20,9 +20,13 @@
 import type {Mode} from "./types";
 
 type Font = "main" | "ams"
+// Some of these have a "-token" suffix since these are also used as `ParseNode`
+// types for raw text tokens, and we want to avoid conflicts with higher-level
+// `ParseNode` types. These `ParseNode`s are constructed within `Parser` by
+// looking up the `symbols` map.
```

Comment:
Does using the same string actually cause an issue?  It seems like they're different types so `flow` should keep them straight for us.  Am I missing something?

Reply:
Yes, `Group` here is a different type, but _every_ string in `Group` is also used as a `ParseNode` type due to [this line in `Parser.js`](https://github.com/Khan/KaTeX/blob/ba24d2073ba44664a3ba82d0d22c309834b60385/src/Parser.js#L968-L969), resulting in this unfortunate name conflict. All of these `Group` values are listed under the second "section" of types within `ParseNode.js` where the value type is `string`. See the comment in this PR in `ParseNode.js` that says

    // From symbol groups, constructed in Parser.js via `symbols` lookup.
    // (Some of these have "-token" suffix to distinguish them from existing
    // `ParseNode` types.)

Reply:
I see.  Yeah, reusing those strings feels bad news.  Thanks for highlighting this issue.